src/libxrpl/tx/paths/PaySteps.cpp

```diff
@@ -53,12 +53,17 @@ checkNear(IOUAmount const& expected, IOUAmount const& actual)
     auto const r = diff / std::max(std::abs(a), std::abs(b));
     return r <= ratTol;
 };
-
 bool
-checkNear(XRPAmount const& expected, XRPAmount const& actual)
+checkNear(MPTAmount const& expected, MPTAmount const& actual)
 {
-    return expected == actual;
-};
+    if (expected == actual)
+        return true;
+    Number ratTol = Number(1, -3);
+    Number const diff = abs(expected.value() - actual.value());
+    auto const r =
+        diff / std::max(std::abs(expected.value()), std::abs(actual.value()));
+    return r <= ratTol;
```

Comment:
The use of auto and abs without a namespace is confusing.
Without unit tests it is very hard to be confident that we are not getting rounding errors.
Minor code cleanup to make the intentions explicit:

```suggestion

    Number const ratTol = Number(1, -3);
    Number const ev = expected.value();
    XRPL_ASSERT(
        (std::is_same_v<decltype(ripple::abs(ev)), Number>),
        "ripple::checkNear: abs(ev) must return Number");

    Number const av = actual.value();
    XRPL_ASSERT(
        (std::is_same_v<decltype(ripple::abs(av)), Number>),
        "ripple::checkNear: abs(av) must return Number");

    Number const diff = ripple::abs(ev - av);
    Number const denom = std::max<Number>(ripple::abs(ev), ripple::abs(av));
    Number const r = diff / denom;
    return r <= ratTol;
```

Reply:
The first `abs` should have been `std::abs`. I'll change that and I'll declare `r` explicitly as `Number`.  I don't see rationale for checking the type. The compile would have failed has it not been possible to convert to `Number`.  `ratTol` and `diff` are explicit `Number`. So all the math is in `Number`. I'm not sure why would there be rounding errors. Can you clarify?

Reply:
Yes, I agree if everything is explictly a Number -  what you did is enough